### PR TITLE
fix: add native function to create preload script

### LINKED ISSUE #9276: preload script cannot be loaded if Content Security Policy is enabled
* Electron version: 1.6.5
* Operating system: macOS

### Expected behavior

Preload script can be loaded even if CSP is enabled

### Actual behavior

Preload script is prevented from being loaded by CSP

### How to reproduce

I could not prepare server with CSP, so let me show the sequence to reproduce with my app...

1. Download v0.2.2 from https://github.com/rhysd/Mstdn/releases
2. Unarchive it and copy [this config.json](https://gist.github.com/rhysd/9f656d4e93f40fe2e41752dfbe08cdc5) to `~/Application Support/Mstdn/config.json`
3. Open app with `$ NODE_ENV=development /path/to/Mstdn.app/Contents/MacOS/Mstdn`
4. It will open DevTools for browser window. Console tab should show the error.

In my app, browser window is opened with `nodeIntegration: false` and `preload: '...'` web preferences for external URL https://mstdn.jp where CSP is enabled.

Above sequence show following error.

It says that `eval` cannot be used because of CSP. The `eval` is used at VM28:784 for loading preload script specified with `preload: '...'` option on creating `BrowserWindow` instance.

Following is a screenshot of the code:

Preload script is loaded with `eval()` function but it is blocked. Is it possible to remove the check by CSP while loading preload script?
Preload script is a way for loading some user script even if Node integration is disabled for external browser AFAIK. So, if it is blocked by CSP, preload script can no longer used for injecting some code to external page safely.

### PR DESCRIPTION
**Pull request by PalmerAL**
Fixes #9276.

It looks like ```file:///``` and ```data:``` URL's are always able to use eval() regardless of what the CSP is, so I'm not really sure how to add a test for this. I guess a custom protocol might work?

### CODE REVIEW

File: spec/webview-spec.js [hunk 1]
```diff
@@ -248,6 +248,37 @@ describe('<webview> tag', function () {
       })
     })
 
+    it('runs in the correct context', async () => {
```
Comment (PalmerAL) [inline spec/webview-spec.js]: I didn't see an existing test for the unsandboxed version, but maybe there's one I've missed?

Comment (tarruda) [reply spec/webview-spec.js]: Unsandboxed preload exposes everything by default so it is not needed.

Comment (tarruda) [reply spec/webview-spec.js]: Technically, the test title should be "run in the correct scope". Context denotes the global JS environment, and both the page and preload run in the same context.

**Comment by PalmerAL** [comment]
@tarruda can this be merged?

**Comment by tarruda** [comment]
@PalmerAL I believe so, but I don't have permissions to merge it. Can you push again to trigger a new CI run? Maybe after all tests are passing it will be merged.

**Comment by PalmerAL** [comment]
@zcbenz (or someone else) can this be merged?

**Pull request merged**